### PR TITLE
TBOX-141: Add workflow for release branch creation

### PR DESCRIPTION
**Pull request by keziah-tamr**
# ↪️ Pull Request

Add a new workflow for the creation of a release branch and the creation of 2 PRs (one against main and one against develop)

## ✔️ PR Todo

- [ ] Add documentation
- [ ] Add examples
- [ ] Ensure you are following the practices in the [contributor guide](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit) and [coding_standards](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards)
  * [Use consistent and descriptive name conventions](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.477s36w4rds9) 
  * [Uses google-style docstrings](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Google-style-docstrings)
  * [Uses versioned API/client methods only](https://docs.google.com/document/d/1c_hhpDMhO0zN793Z_HFOz42Q4g7l1nPRVQ5pXOUsMJU/edit#heading=h.yv6df99fyrq2)
  * [Avoid global variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-global-variables)
  * [Avoid giant try/catch](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Avoid-giant-try-/-except)
  * [Don't use mutable default variables](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Don%E2%80%99t-use-mutable-default-arguments)
  * [Use a `main()` function in your scripts](https://datatamr.atlassian.net/wiki/spaces/FEKB/blog/2020/04/14/1137606830/Customer+Success+Coding+Standards#Use-a-%E2%80%9Cmain()%E2%80%9D--function-in-your-scripts)
- [ ] Added/updated testing for this change (ensure you are testing any changes/new code!)
- [ ] Update title to link to TBOX issue (i.e. start pull request title with TBOX-xx) 
- [ ] Approval from first code-reviewer
- [ ] Approval from second code-reviewer (only required if a second reviewer has left comments)
- [ ] Pass all checks

### CODE REVIEW

**Comment by skalish** [comment]
This LGTM, as does the result assuming this is what was used to set up the branch 'release-4.5.6-toolname` on your fork.

Before I click approve, I want to make sure I have the right understanding of what this does. Is any of the following incorrect:
To clarify what this PR does:
- Creates a branch off of `develop` with the name `release-<version_number>`
- Opens a PR from `release-<version_number>` to `main`
- Opens a PR from `release-<version_number>` to `develop`
So after this, `release-<version_number>` can be manually prepared for release, and then the PRs can be manually closed via merge into both target branches.

The "short-version" number is only used for building the docs.

**Comment by keziah-tamr** [comment]
@skalish your summary is correct

**Review by skalish — APPROVED** [review]
LGTM

**Pull request merged**
